daemon.go

```diff
@@ -41,7 +46,7 @@ func (d *menderDaemon) shouldStop() bool {
 func (d *menderDaemon) Run() error {
 	// figure out the state
 	for {
-		state, cancelled := d.mender.GetState().Handle(d.mender)
+		state, cancelled := d.mender.RunState(&d.sctx)
```

Comment:
This looks a bit fishy. In real OO language I would write we are calling a method on an object which we are passing as an argument to itself (not explicitly, but `mender` is part of `sctx`).